### ISSUE TITLE
Items from stash are dropped

### ISSUE THREAD

**Issue by Puros** [flareteam/flare-engine#748]
When ctrl-clicking an item in the stash, it is dropped to the ground if inventory is full.  The same happens when taking coins from the stash as follows:

Issue #1:

1) Fill all inv slots including one slot with a partial stack of coins.
2) Ctrl-click (or drag) a full stack of coin from the stash

The excess coins will fall to the ground rather than remaining in the stash.

Issue #2:
1) Fill all inv slots
2) Ctrl-click an item in the stash

The item is dropped to the ground rather than remaining in the stash.

If the object of this behavior is to allow clearing a stash slot when inventory is full, then dropping something from the stash on the ground should do it.  So...

Issue #3:

1) Drag an item from the stash and drop it onto the play field.

Unlike the behavior when this is done from inventory, the item is not dropped to the ground but is returned to the stash.

TL;DR
- Drag and drop items from stash to ground
- Return dragged items (including _partial_ coin stacks) if inventory has no room for them
